### DISCUSSION TITLE
Recompile lwip libraries in Windows - Step by step

### DISCUSSION THREAD

**Discussion by vpapanik** [esp8266/Arduino#8616] [Show and tell]
Install **Git** for windows (https://git-scm.com/download/win)

Install **MinGW** for windows (https://sourceforge.net/projects/mingw)

In **MinGW** installation manager, enable:

- mingw-developer-toolkit
- mingw32-base
- mingw32-gcc-g++
- msys-base

_(not sure if all are necessary but just in case)_

put _C:\MinGW\bin_ in Windows PATH (check this https://www.rapidee.com)

**reboot PC**
--------------------

rename **C:\MinGW\bin\mingw32-make.exe** to **C:\MinGW\bin\make.exe**

open **bash** terminal (just type bash in run)

_(from now on assume that hardware files are in \Documents\Arduino\hardware\esp8266com\esp8266)_

go to _\tools\sdk\lwip2_

optional: change **lwip options** in _\tools\sdk\lwip2\builder\glue-lwip\arduino\lwipopts.h_

(caution : **NOT** _\tools\sdk\lwip2\include\lwipopts.h_, this will be overwritten !)

type in bash: **make install** to build libraries in _\esp8266com\esp8266\tools\sdk\lwip2\builder_

type in bash: **make clean** to clean _\esp8266com\esp8266\tools\sdk\lwip2\builder_

finally copy new libraries to _\esp8266com\esp8266\tools\sdk\lib_

**Comment by mcspr** [comment]
Also note that https://github.com/d-a-v/esp82xx-nonos-linklayer has cmake support, see README-CMake.md there
Readme here could also use an update about the dependencies, at least for common systems
